File: Tutorial/JSProblems/secondLargest.js


function secondLargest(arr){

  let firstLargest = 0;
  let secondLargest = 0;

  for( let i = 0; i < arr.length; i++){

    if( arr[i] > firstLargest ){
      secondLargest = firstLargest;
      firstLargest = arr[i];
      
    }
   else if( arr[i] > secondLargest && arr[i] != firstLargest){
      secondLargest = arr[i];
    }
  }
  return secondLargest;
}

console.log(secondLargest([3, 5, 2, 9, 6, 10, -2]));

File: Tutorial/javaProblems/SecondLargest.java

public class SecondLargest{

  public static int findSecondLargest(int[] array){

    int largest = array[0];
    int secondLargest = largest;
    
    for(int i = 1; i < array.length; i++){
       if( array[i] > largest){
        secondLargest = largest;
        largest = array[i];
    }
    else if( array[i] > secondLargest && array[i] != largest){
      secondLargest = array[i];
    }
  }

  return secondLargest;
}

  public static void main(String[] args) {

    int[] arr = new int[]{3, 2, 6, 5, 4, 9, 6};
    System.out.println(findSecondLargest(arr));
    
  }
  
}